driver-core/src/test/java/com/datastax/driver/core/policies/ErrorAwarePolicyTest.java

```diff
@@ -190,6 +226,7 @@ public void should_exclude_host_after_reaching_inclusion_threshold() throws Inte
 
         try {
             Session session = cluster.connect();
+            cluster.register(latencyTracker);
```

Comment:
I agree, it just happens to work that way because `CopyOnWriteArraySet` is backed by `CopyOnWriteArrayList`, which is insertion-ordered but I don't think the Set being insertion ordered is explicitly guaranteed, so this can break at any time.  Based on our recent discussion I'll write a separate set of unit tests that more explicitly verify the contract of `ErrorAwarePolicy` by working with the `PerHostErrorTracker` and its meters directly.  I think I will keep the integration tests because they validate things end to end, but make it explicit in documentation that they are fragile and make it clear what it's assumptions/dependencies are about the internal implementation.